README
NAME
    EWS::Client - Microsoft Exchange Web Services Client

VERSION
    version 1.143070

SYNOPSIS
    Set up your Exchange Web Services client.

     use EWS::Client;
     use DateTime;
 
     my $ews = EWS::Client->new({
         server      => 'exchangeserver.example.com',
         username    => 'oliver',
         password    => 's3krit', # or set in $ENV{EWS_PASS}
     });

    Then perform operations on the Exchange server:

     my $entries = $ews->calendar->retrieve({
         start => DateTime->now(),
         end   => DateTime->now->add( months => 1 ),
     });
 
     print "I retrieved ". $entries->count ." items\n";
 
     while ($entries->has_next) {
         print $entries->next->Subject, "\n";
     }
 
     my $contacts = $ews->contacts->retrieve;

DESCRIPTION
    This module acts as a client to the Microsoft Exchange Web Services API.
    From here you can access calendar and contact entries in a nicely
    abstracted fashion. Query results are generally available in an iterator
    and convenience methods exist to access the properties of each entry.

AUTHENTICATION
    Depending on the configuration of the Microsoft Exchange server, you can
    use either HTTP Basic Access Auth, or NTLM Negotiated Auth, from this
    module. The default is HTTP Basic Access Auth, so if using NTLM, the
    following additional option to "new()" is required:

     use_negotiated_auth => 1,

METHODS
  EWS::Client->new( \%arguments )
    Instantiates a new EWS client. There won't be any connection to the
    server until you call one of the calendar or contacts retrieval methods.

    "server" => Fully Qualified Domain Name (required)
        The host name of the Exchange server to which the module should
        connect.

    "username" => String (required)
        The account username under which the module will connect to
        Exchange.

        For Basic Access Auth this value will be URI encoded by the module,
        meaning you don't have to worry about escaping any special
        characters. For NTLM Negotiated Auth, pass a "user@domain" format
        username and it will automatically be converted into Windows'
        "domain\user" format for you.

    "password" => String OR via $ENV{EWS_PASS} (required)
        The password of the account under which the module will connect to
        Exchange.

        For Basic Access Auth this value will be URI encoded by the module.
        You can also provide the password via the "EWS_PASS" environment
        variable.

    "use_negotiated_auth" => True or False value
        The module will assume you wish to use HTTP Basic Access Auth, in
        which case you should enable that in your Exchange server. However
        for negotiated methods such as NTLM set this to a True value.

    "schema_path" => String (optional)
        A folder on your file system which contains the WSDL and two further
        Schema files (messages, and types) which describe the Exchange 2007
        Web Services SOAP API. They are shipped with this module so your
        providing this is optional.

    "server_version" => String (optional)
        In each request to the server is specified the API version we expect
        to use. By default this is set to "Exchange2007_SP1" but you have
        the opportunity to set it to "Exchange2007" if you wish using this
        option.

  $ews->calendar()
    Retrieves the EWS::Client::Calendar object which allows search and
    retrieval of calendar entries and their various properties. See that
    linked manual page for more details.

  $ews->contacts()
    Retrieves the EWS::Client::Contacts object which allows retrieval of
    contact entries and their telephone numbers. See that linked manual page
    for more details.

  $ews->folders()
    Retrieves the EWS::Client::Folder object which allows retrieval of
    mailbox folder entries and their sizes. See that linked manual page for
    more details.

  $ews->dls()
    Retrieves the EWS::Client::DistributionList object which allows
    retrieval of distribution list entries and their email addresses and
    names. See that linked manual page for more details.

KNOWN ISSUES
    *   No handling of time zone information, sorry.

    *   The "SOAPAction" Header might be wrong for Exchange 2010.

THANKS
    To Greg Shaw for sending patches for NTLM Authentication support and
    User Impersonation.

AUTHOR
    Oliver Gorwits <oliver@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2014 by University of Oxford.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
1.200001  2015-06-14

  * Fix typo in POD (T. Hubbard)

1.200000  2015-06-14

  * Nothing to report.

1.143070  2014-11-03 21:06:51 Europe/London

  * [#5] ExpandDL support for expanding distribution lists (FunkyShu)

1.141040  2014-04-14 21:05:15 Europe/London

  * Retrieve other user calendar items (retupmoca)

1.131710_001 2013-06-20 23:07:20 Europe/London

  * Retrieval of mailbox folder entries and their sizes (G. Shaw)
  * Calendar invitees will show name and routing in addition to email address (G. Shaw)

1.130570  2013-02-26 20:15:48 Europe/London

  * Next stable release.

1.130540_002 2013-02-23 16:47:07 Europe/London

  * Docs for new properties.

1.130540_001 2013-02-23 16:05:31 Europe/London

  * Add JobTitle, CompanyName, BusinessHomePage, EmailAddresses, and PhysicalAddresses
    (Ron Tomassini)

1.113000  2011-10-27 19:47:48 Europe/London

  * No longer attempt naive interpretation of Phone Numbers (M. Redinger)

1.111982  2011-07-17 16:48:37 Europe/London

  * Need to re-release to fix botched deps

1.111981  2011-07-17 13:18:50 Europe/London

  * Add dependencies for HTTPS and NTLM support, always
  * Added example script

1.111970  2011-07-16 17:19:24 Europe/London

  * Use different Dist::Zilla author bundle

1.111940  2011-07-13 19:59:27 Europe/London

  * Fixes for NTLM Authentication (code and docs)

1.110911  2011-04-01 15:04:00 Europe/London

  * Apply the patch correctly this time.

1.110910  2011-04-01 14:27:58 Europe/London

  * Fixed a couple of doc typos (jaybuff via github)

1.103620  2010-12-28 00:15:12 Europe/London

  * Fix misspelt CalendarType (rt.cpan #56188)
  * Add note about possible Exchange 2010 issues (rt.cpan #55890)
  * Add server_version attribute (rt.cpan #61914)
  * Add support for NTLM Authentication (rt.cpan #56189 - Greg Shaw)
  * Add support for Impersonation (rt.cpan #56189 - Greg Shaw)

1.103610  2010-12-27 18:29:05 Europe/London

  * Port to Dist::Zilla

libews-client-perl (0.06) UNRELEASED; urgency=low

  * Add check for no items returned from server - thanks Mike M. 

 -- Oliver Gorwits <oliver.gorwits@oucs.ox.ac.uk>  Mon, 22 Mar 2010 21:25:58 +0000

libews-client-perl (0.05) UNRELEASED; urgency=low

  * Added POD note about HTTP BasicAuth - thanks to Darren W. 

 -- Oliver Gorwits <oliver.gorwits@oucs.ox.ac.uk>  Tue, 16 Mar 2010 19:48:47 +0000

libews-client-perl (0.04) UNRELEASED; urgency=low

  * Add Contacts retrieval.

 -- Oliver Gorwits <oliver.gorwits@oucs.ox.ac.uk>  Sun, 14 Mar 2010 22:59:03 +0000

libews-client-perl (0.03) UNRELEASED; urgency=low

  * Result set is an iterator, add MooseX::Iterator.

 -- Oliver Gorwits <oliver.gorwits@oucs.ox.ac.uk>  Tue, 23 Feb 2010 10:48:57 +0000

libews-client-perl (0.02) UNRELEASED; urgency=low

  * Add File::ShareDir dependency.

 -- Oliver Gorwits <oliver.gorwits@oucs.ox.ac.uk>  Mon, 22 Feb 2010 21:43:19 +0000

libews-client-perl (0.01) UNRELEASED; urgency=low

  * Initial release on an unsuspecting world.

 -- Oliver Gorwits <oliver.gorwits@oucs.ox.ac.uk>  Mon, 22 Feb 2010 19:01:46 +0000


Makefile.PL

use strict;
use warnings;

use ExtUtils::MakeMaker 6.30;

use File::ShareDir::Install;
install_share dist => "share";

my %WriteMakefileArgs = (
  'ABSTRACT' => 'Microsoft Exchange Web Services Client',
  'AUTHOR' => 'Oliver Gorwits <oliver@cpan.org>',
  'BUILD_REQUIRES' => {
    'Test::More' => '0'
  },
  'CONFIGURE_REQUIRES' => {
    'ExtUtils::MakeMaker' => '6.30',
    'File::ShareDir::Install' => '0.03'
  },
  'DISTNAME' => 'EWS-Client',
  'EXE_FILES' => [],
  'LICENSE' => 'perl',
  'NAME' => 'EWS::Client',
  'PREREQ_PM' => {
    'Authen::NTLM' => '0',
    'Carp' => '0',
    'Data::Dumper' => '0',
    'DateTime' => '0',
    'DateTime::Format::ISO8601' => '0',
    'Encode' => '0',
    'File::ShareDir' => '0',
    'HTML::Strip' => '0',
    'LWP::Authen::Ntlm' => '0',
    'LWP::Protocol::https' => '6.02',
    'Log::Report' => '0',
    'Moose' => '0',
    'Moose::Role' => '0',
    'Moose::Util::TypeConstraints' => '0',
    'MooseX::Iterator' => '0',
    'URI::Escape' => '0',
    'XML::Compile::SOAP11' => '0',
    'XML::Compile::Transport::SOAPHTTP' => '0',
    'XML::Compile::WSDL11' => '0'
  },
  'VERSION' => '1.200001',
  'test' => {
    'TESTS' => 't/*.t'
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);

package
MY;
use File::ShareDir::Install qw(postamble);


